### DISCUSSION TITLE
Get this error at startup, on Windows 10

### DISCUSSION THREAD

**Discussion by trobbins1981** [johnlindquist/kit#960] [Errors]
Unhandled Promise Rejection
TypeError: (o || []).map is not a function
    at C:/Users/trobb/AppData/Local/Programs/kit/resources/app.asar/main.prod.js:2:306418
    at u (C:/Users/trobb/AppData/Local/Programs/kit/resources/app.asar/main.prod.js:2:281293)
    at Generator._invoke (C:/Users/trobb/AppData/Local/Programs/kit/resources/app.asar/main.prod.js:2:281046)
    at Generator.next (C:/Users/trobb/AppData/Local/Programs/kit/resources/app.asar/main.prod.js:2:281656)
    at Gl (C:/Users/trobb/AppData/Local/Programs/kit/resources/app.asar/main.prod.js:2:287231)
    at a (C:/Users/trobb/AppData/Local/Programs/kit/resources/app.asar/main.prod.js:2:287435)
    at C:/Users/trobb/AppData/Local/Programs/kit/resources/app.asar/main.prod.js:2:287496
    at new Promise (<anonymous>)
    at C:/Users/trobb/AppData/Local/Programs/kit/resources/app.asar/main.prod.js:2:287375
    at C:/Users/trobb/AppData/Local/Programs/kit/resources/app.asar/main.prod.js:2:307286

**Comment by johnlindquist** [comment]
@trobbins1981 Is this the very first time opening the app? And it happens before any UI appears and just crashes?

**Reply by trobbins1981** [reply]
It does happen the very first time I open the app.  The UI briefly shows then disappears and the error message pops up

**Reply by johnlindquist** [reply]
Can you please check for and post any logs you find here:

%USERPROFILE%\AppData\Roaming\Kit\logs\main.log
%USERPROFILE%\AppData\Roaming\Kit\logs\update.log
%USERPROFILE%\.kit\logs\kit.log

**Reply by trobbins1981** [reply]
[main.log](https://github.com/johnlindquist/kit/files/10141425/main.log)
[update.log](https://github.com/johnlindquist/kit/files/10141427/update.log)

**Reply by johnlindquist** [reply]
@trobbins1981 thanks, hmm...

Do you have a ~/.kit/db/scripts.json file?

Please post it if it's there

**Reply by trobbins1981** [reply]
[scripts.zip](https://github.com/johnlindquist/kit/files/10141767/scripts.zip)

**Reply by johnlindquist** [reply]
Hmm, everything looks OK... Would you mind trying the latest nightly/pre-release from here: https://github.com/johnlindquist/kitapp/releases/tag/v1.36.60

Let me know if it throws the same error? Thanks a million

**Reply by trobbins1981** [reply]
I updated and I get the same error

**Reply by johnlindquist** [reply]
Sorry, was there a log here?

%USERPROFILE%\.kit\logs\kit.log

**Reply by trobbins1981** [reply]
[kit.log](https://github.com/johnlindquist/kit/files/10142087/kit.log)

**Reply by johnlindquist** [reply]
I'm seeing 'second-instance' in your log. Can you try checking the Task Manager for a stray "Kit" process and make sure to end all of them. If you find one, try opening again?

**Reply by trobbins1981** [reply]
there were 6 kit processes running. If i shut down all of them, 'Ctrl+;' no longer works, which is the documented way to launch this app.  If i launch via the exe file, then i have 6 processes again running and i get the same error when doing Ctrl+;

**Reply by johnlindquist** [reply]
Oh, sorry, I was misunderstanding. I thought it was crashing during the launch/initial install process then exiting.

Can you try typing:

```
kit,,
```

Anywhere on your system to see if it will run a Snippet script? It "should" expand it to: `https://scriptkit.com`

**Reply by trobbins1981** [reply]
when I type that it replaces it with an empty string

**Reply by johnlindquist** [reply]
Please type `kit,,` again, then send me the kit.log. It may provide some more insight since it's trying to run a script

**Reply by trobbins1981** [reply]
[kit.log](https://github.com/johnlindquist/kit/files/10142365/kit.log)

**Reply by johnlindquist** [reply]
Okay, this is kind of a big ask...

In the file `~/.kit/api/kit.js`

Search for:

```
let p = global.sendWait(Channel.SET_CHOICES, choices);
```

And add a log line above it so it looks like:

```
log({choices})
let p = global.sendWait(Channel.SET_CHOICES, choices);
```

Then run the `cmd+;` shortcut a ~5 times and send me the kit.log again.

**Comment by trobbins1981** [comment]
[kit.log](https://github.com/johnlindquist/kit/files/10142463/kit.log)

i didn't see that exact line, as maybe you are looking at the typescript version.  But, i made my best guess, restarted Kit , did the command 5 times, and sent the log

**Reply by johnlindquist** [reply]
It looks like the error is somehow involved with formatting the scripts.json prior to displaying them with the main shortcut. Let's try a script that doesn't require listing the scripts:

Create a file here:

```
~/.kenv/scripts/testing-shortcut.js
```

With these contents:

```
// Name: Testing Shortcut
// Shortcut: ctrl shift i

import "@johnlindquist/kit"

await editor(`Is this working?`)
```

Save it, then hit `ctrl+shift+i` on your keyboard to see if it launches?

**Reply by trobbins1981** [reply]
nothing happens and their are no logs. I'm guessing my OS uses that shortcut for something else

**Reply by johnlindquist** [reply]
Oh, then pick another shortcut and put it in the `// Shortcut: ctrl shift whatever` and give it another try

**Reply by trobbins1981** [reply]
nothing happens still, but I do see logs
[kit.log](https://github.com/johnlindquist/kit/files/10143257/kit.log)

**Reply by johnlindquist** [reply]
Ok, thanks so much for all your help. I'll dig/debug more and get back to you later

**Reply by johnlindquist** [reply]
I pushed out a new build pre-release:

https://github.com/johnlindquist/kitapp/releases/tag/v1.37.1

Would you mind trying that and sending the kit.log again?

**Reply by trobbins1981** [reply]
with new release

![image](https://user-images.githubusercontent.com/3045359/205681116-9b4006ba-d15d-4027-9b76-12f31376719f.png)
[kit.log](https://github.com/johnlindquist/kit/files/10155660/kit.log)

**Reply by johnlindquist** [reply]
@trobbins1981 can you test the Windows build from here? I made significant changes to the way it builds which should fix the issue:

https://github.com/johnlindquist/kitapp/releases/tag/v1.40.1

**Comment by jbrown123** [comment]
Same here. Brand new first time install on Win 10 and I get the Unhandled promise rejection. This is after the KIT SETUP UI is displayed, as it is going through install steps. Specifically it seems to be trying to execute the .kenv .env setup.

Unhandled Promise Rejection
Error: spawn C:/Users/[redacted]/AppData/Local/Programs/kit/resources/fallbacks/windows/clipboard_x86_64.exe ENOENT
    at r (C:/Users/[redacted]/AppData/Local/Programs/kit/resources/app.asar/main.prod.js:2:543688)
    at n (C:/Users/[redacted]/AppData/Local/Programs/kit/resources/app.asar/main.prod.js:2:543859)
    at ChildProcess.e.emit (C:/Users/[redacted]/AppData/Local/Programs/kit/resources/app.asar/main.prod.js:2:544000)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:291:12)

Using Kit-Windows-1.40.62-x64.exe

**Reply by johnlindquist** [reply]
@jbrown123 Can you try this comment please and let me know? 🙏

https://github.com/johnlindquist/kit/issues/1007#issuecomment-1400807465

**Reply by jbrown123** [reply]
The obvious problem is that the "fallbacks" directory doesn't exist.

```
 Directory of c:\Users\[redacted]\AppData\Local\Programs\kit\resources

01/23/2023  11:54 AM    <DIR>          .
01/23/2023  11:54 AM    <DIR>          ..
01/07/2023  07:31 PM        38,440,139 app.asar
01/07/2023  07:31 PM    <DIR>          app.asar.unpacked
01/07/2023  07:31 PM               159 app-update.yml
01/23/2023  11:55 AM    <DIR>          assets
01/07/2023  07:31 PM           123,600 elevate.exe
               3 File(s)     38,563,898 bytes
```
